Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml

```diff
@@ -0,0 +1,237 @@
+      | where DeviceEventCategory == "AUDIT" 
+          and (toupper(ProcessName) !in ("LOGIN", "LOGOUT", "UNSUCCESSFUL_LOGIN"))
+      | parse trim(@'[^\w(")]+', Message) with * "Reason=" Reason:string 
+      | extend Reason = trim('"',Reason)
+      | extend 
+          EventResultDetails = Reason,
+          severity = toint(LogSeverity)
+      | lookup SeverityLookup on severity
+      | lookup EventTypeLookup on $left.EventOutcome == $right.ChangeType_s
+      | lookup ObjectTypeLookup on $left.FileType == $right.ObjectType_s
+      | extend
+          EventResult = "Success", 
+          EventSchema = "AuditEvent",
+          EventSchemaVersion = "0.1.0",
+          EventVendor = "Barracuda",
+          EventProduct = "WAF",
+          EventCount = toint(1)
+      | extend
+          EventType = EventType_lookup,
+          Dvc = DeviceName, 
+          EventStartTime = iff(isnotempty(FlexNumber2), unixtime_milliseconds_todatetime(tolong(ReceiptTime)-tolong(FlexNumber2)), unixtime_milliseconds_todatetime(tolong(ReceiptTime))),
+          Operation = ProcessName,
+          DvcIpAddr = DeviceAddress,
+          NewValue = DeviceCustomString1,
+          SrcIpAddr = SourceIP,
+          EventMessage = Message,
+          OldValue = DeviceCustomString2,
+          DvcHostname = DeviceName,
+          ActorUsername = DestinationUserName,
+          Object = FileName,
+          ThreatConfidence = toint(ThreatConfidence)     
+      | extend
+          Src = SrcIpAddr,
+          EventEndTime = EventStartTime,
+          ActorUsernameType = iff(isnotempty(ActorUsername), "Simple", ""),
+          ActorUserType = iff(isnotempty(ActorUsername), "Admin", ""),
+          User = ActorUsername,
+          Value = NewValue 
+      | extend
+          IpAddr = SrcIpAddr,
+          ValueType = iff(isnotempty(Value), "Other", "")
+      | project-away
+          EventType_lookup,
+          ThreatConfidence,
+          CommunicationDirection,
+          AdditionalExtensions,
+          Device*,
+          Source*,
+          Reason,
+          Destination*,
+          Activity,
+          LogSeverity,
+          ApplicationProtocol,
+          ProcessID,
+          ExtID,
+          Protocol,
+          ReceiptTime,
+          SimplifiedDeviceAction,
+          OriginalLogSeverity,
+          ProcessName,
+          EndTime,
+          ExternalID,
+          File*,
+          ReceivedBytes,
+          Message,
+          Old*,
+          EventOutcome,
+          Request*,
+          StartTime,
+          Field*,
+          Flex*,
+          Remote*,
+          Malicious*,
+          severity,
+          ThreatSeverity,
+          IndicatorThreatType,
+          ThreatDescription,
+          _ResourceId,
+          SentBytes,
+          ReportReferenceLink,
+          Computer,
+          TenantId
+  }
+  ;
+  union isfuzzy = true 
+      BarracudaCustom(disabled=disabled),
+      BarracudaCEF(disabled=disabled)
```

Comment:
@jayeshprajapaticrest  Why we added filtering to these functions separately?  We should do union under a parent function and add filter to that parent function. Please refer this : https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

Reply:
Done